docs/gallery/tiled-random-samples.ipynb

```diff
@@ -95,6 +95,11 @@
    "metadata": {},
    "outputs": [],
    "source": [
+    "# Adjust some of plots\n",
+    "plots['uniform'].ax.set_ylim(0, 0.11)\n",
+    "plots['beta'].ax.set_ylim(0, 0.2)\n",
+    "plots['triangular'].ax.set_ylim(0, 0.2)\n",
+    "\n",
```

Comment:
At some point we should look at the performance of `tiled` because making this figure is super slow :-( 
It was designed for making a small number of subplots, but you're right that this is probably the first thing people will try to make with it.

Reply:
Yes... it takes about 40 seconds in my machine.
But I thought it was not so important,
cuz you can just use ``matplotlib`` subplots and manually pass ``ax`` to the plotting function if it should be faster.

Do you know why it is so slow...?

Reply:
> Do you know why it is so slow...?

I've done a bit of digging. There are a few reasons:

1. Plotting with plopp is just slower (about 3x slower) than pure matplotlib. We do a bunch of preprocessing on the data, converting it to a data array, checking units etc. We also construct more things such as a View, Canvas, Figure, all of which are Plopp-specific objects.

2. The (lazy) way tiled is currently implemented is that every figure is made twice. When you assign a plot to e.g. `tiled[0, 0]`, you first make the plot, and then assign it. When you assign it, there is an operation (`copy_figure`) where a new figure is made from the old one. So you make it twice.

What's kind of strange is that, if I make the subplots manually with matplotlib, and then just iterate over the plots and assign axes to each plot, it takes 30s instead of about 1min using Tiled on my laptop.

But just making the matplotlib subplots and plotting some numpy data and setting axes labels only takes 1s. So there is a factor of 30, not a factor of 3 which is what I saw for just making a single plot.

Reply:
What's even more strange is that making the same plot in a script and not in jupyter only takes 8 seconds...

Reply:
I found the culprit: [these lines](https://github.com/scipp/plopp/blob/main/src/plopp/backends/matplotlib/utils.py#L38-L40) are switching to another backend and then switching back when in jupyter. This seems to get slower and slower the larger the figure is (or maybe the more plots there are?).

I cannot remember why I needed this but I am guessing there was a good reason, will try to figure it out.

Reply:
Then I'll just merge it since it's not affecting docs building time : D...!